### PR TITLE
Code-gen addcdiv again

### LINKED ISSUE #4213: Seeing a roughly 25% slow down with torch-xla=1.13 because of the addcdiv_ and addcmul_ ops
## 🐛 Bug

For BERT large model, we are seeing around 25% slow down with the latest torch-xla=1.13. The slow down seems to happen because of the addcdiv_ and addcmul_ operations transferring scalars synchronously to device. These are the operations which causes slow down:
1. `exp_avg_sq.mul_(beta2).addcmul_(grad, grad, value=1.0 - beta2)`
2. ` p.data.addcdiv_(exp_avg, denom, value=-step_size)`
(Taken from: https://github.com/huggingface/transformers/blob/v4.6.0/src/transformers/optimization.py#L346)
Note, here the scalars are not special scalars as the value is not 0 or 1 as in this case: https://github.com/pytorch/xla/pull/3953

As the model size grows the number of scalars transfered to device also grows and we end up with a profile that looks like this:

Screen Shot 2022-11-17 at 9 19 59 AM
## To Reproduce
Run the following example:

```
t = torch.randn(1, 3)
t1 = torch.randn(3, 1)
t2 = torch.randn(1, 3)
t.addcdiv_(t1, t2, value=0.1) 
xm.mark_step()
```

This should produce a transfer of scalar synchronously.

1.
4.
5.

## Expected behavior

We didn't see such a transfer with torch-xla 1.12. 

## Environment

 - Reproducible on XLA backend [CPU/TPU]:
 - torch_xla version: 1.13

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
This pull request redoes the addcdiv and addcmul code-gen, and adds a test case to verify that if we reuse the DataCache for scalars.

This needs pytorch/pytorch#92066 to function.

Test Plan:
PJRT_DEVICE=CPU python test/test_operations.py -v -k test_cached_addcdiv

Fixes #4213.

### CODE REVIEW

File: test/test_operations.py [hunk 1]
```diff
@@ -1893,6 +1893,21 @@ def test_sigmoid_bounds(self):
...
+  def test_cached_addcdiv(self):
+    xla_device = xm.xla_device()
+    met.clear_all()
+
+    t1 = torch.randn(1, 3).to(xla_device)
+    t2 = torch.randn(1, 3).to(xla_device)
+    t3 = torch.randn(1, 3).to(xla_device)
+    t1.addcdiv_(t2, t3, value=0.1)
+    # The following two scalars shouldn't trigger TransferToServerTime.
+    t1.addcdiv_(t2, t3, value=0.1)
```
Comment (JackCaoG) [inline test/test_operations.py:1905]: Let's may this test a bit more explicit by checking the `met.metric_data("TransferToServerTime")[0]` before second `addcdiv_` and after second `addcdiv_` to verify that second `addcdiv_` does not introduce new transfer.

Comment (alanwaketan) [reply test/test_operations.py:1905]: Okay, let me do that.

**Review by JackCaoG — APPROVED** [review]
Thanks!

**Comment by alanwaketan** [comment]
Thanks Jack for approving.

**Review by wonjoolee95 — APPROVED** [review]

**Pull request merged**